File: ordersync-payment/Services/PaymentService.cs
using ordersync_payment.Models;
using System.Text;
using System.Text.Json;
using System.Net.Http;

namespace ordersync_payment.Services;

public class PaymentService
{
    private static List<Payment> payments = new();

    public async Task<Payment> ProcessPayment(long orderId, decimal amount)
    {
        // Simulación de pago
        var payment = new Payment
        {
            Id = payments.Count + 1,
            OrderId = orderId,
            Amount = amount,
            Status = amount > 0 ? "approved" : "failed"
        };

        payments.Add(payment);
        
        // webhook
        using var httpclient = new HttpClient();
        var webhookUrl = "http://localhost:3000/webhook/payment";
        var payload = JsonSerializer.Serialize(payment);
        var content = new StringContent(payload, Encoding.UTF8, "application/json");

        await httpclient.PostAsync(webhookUrl, content);

        return payment;
    }
}

File: ordersync-api/src/services/payment.service.ts
import axios from "axios";

const PAYMENT_API = "http://localhost:5084/payments";

export const processPayment = async (orderId: number, amount: number) => {
  try {
    console.log("Sending to payment service:", {
      OrderId: orderId,
      Amount: amount
    });

    const response = await axios.post(
      PAYMENT_API, 
      {
        OrderId: orderId,
        Amount: amount
      },
      {
        headers: {
          "Content-Type": "application/json",
        },
      }
    );

    console.log("Payment service response:", response.data);

    return response.data;

  } catch (error: any) {
    console.error("Error calling payment service:");
    console.log(error.response?.data || error.message);
    
    throw new Error("Payment service failed");
  }
};